arch/risc-v/src/mpfs/mpfs_head.S

```diff
@@ -100,30 +100,17 @@ __start:
   csrr a0, mhartid
   beqz a0, .skip_e51
 
-  /* Clear sscratch if u54 */
-
-  csrw sscratch, zero
```

Comment:
one simple question: why not use #ifdef/#else/#endif to separate M-mode or S-mode specific code?

Reply:
I like having separate files, the mpfs_head.S file is the M-mode head file, mpfs_shead.S is the S-mode head file. To me this makes sense.